Python/pythonrun.c

```diff
@@ -1047,7 +1047,7 @@ _PyErr_Display(PyObject *file, PyObject *unused, PyObject *value, PyObject *tb)
 
     int unhandled_keyboard_interrupt = _PyRuntime.signals.unhandled_keyboard_interrupt;
 
-    if (!value || PyErr_GivenExceptionMatches(value, PyExc_MemoryError)) {
+    if (!value) {
```

Comment:
Should not it be tested before `PyExceptionInstance_Check(value)`?

Reply:
Actually, reviewing the old code and callees, this is an error condition, so I am adding an assert here at the top